File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        countS = Counter(s)

        countT = Counter(t)

        return countS == countT

        # SC -> O(N)
        # TC -> O(N)
        

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
           

        int[] store = new int[26];


        for(int i = 0; i < s.length();i++){
            store[s.charAt(i) - 'a']++;
            store[t.charAt(i) - 'a']--;
        }

        int[] compare = new int[26];

        return Arrays.equals(store,compare);           

        
    }
}

// SC -> O(26)
// TC -> O(N)